examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart

```diff
@@ -0,0 +1,740 @@
+        style: new TextStyle(
+          color: color == Tab2ConversationBubbleColor.blue
+              ? CupertinoColors.white
+              : CupertinoColors.black,
+          letterSpacing: -0.4,
+          fontSize: 14.0,
+          fontWeight: color == Tab2ConversationBubbleColor.blue
+              ? FontWeight.w300
+              : FontWeight.w400,
+        ),
+      ),
+    );
+  }
+}
+
+class Tab2ConversationAvatar extends StatelessWidget {
+  Tab2ConversationAvatar({this.text, this.color});
+
+  final String text;
+  final Color color;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Container(
+      decoration: new BoxDecoration(
+        shape: BoxShape.circle,
+        gradient: new LinearGradient(
+          begin: FractionalOffset.topCenter,
+          end: FractionalOffset.bottomCenter,
+          colors: <Color>[
+            color,
+            new Color.fromARGB(
+              color.alpha,
+              (color.red - 60).clamp(0, 255),
+              (color.green - 60).clamp(0, 255),
+              (color.blue - 60).clamp(0, 255),
+            ),
+          ],
+        ),
+      ),
+      margin: const EdgeInsets.only(left: 8.0, bottom: 8.0),
+      padding: const EdgeInsets.all(12.0),
+      child: new Text(
+        text,
+        style: const TextStyle(
+          color: CupertinoColors.white,
+          fontSize: 13.0,
+          fontWeight: FontWeight.w500,
+        ),
+      ),
+    );
+  }
+}
+
+List<Widget> buildTab2Conversation() {
+ return <Widget>[
+    new Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget> [
+        new Tab2ConversationBubble(
+          text: "My Xanadu doesn't look right",
+          color: Tab2ConversationBubbleColor.blue
+        ),
+      ],
+    ),
+    new Row(
+      mainAxisAlignment: MainAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: <Widget> [
+        new Tab2ConversationAvatar(
+          text: 'KL',
+          color: const Color(0xFFFD5015),
+        ),
+        new Tab2ConversationBubble(
+          text: "We'll rush you a new one.\nIt's gonna be incredible",
+          color: Tab2ConversationBubbleColor.gray,
+        ),
+      ],
+    ),
+    new Row(
+      mainAxisAlignment: MainAxisAlignment.end,
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget> [
+        new Tab2ConversationBubble(
+          text: 'Awesome thanks!',
+          color: Tab2ConversationBubbleColor.blue,
+        ),
+      ],
+    ),
+    new Row(
+      mainAxisAlignment: MainAxisAlignment.start,
+      mainAxisSize: MainAxisSize.min,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: <Widget> [
+        new Tab2ConversationAvatar(
+          text: 'SJ',
+          color: const Color(0xFF34CAD6),
+        ),
+        new Tab2ConversationBubble(
+          text: "It's because you're holding\nit wrong",
```

Comment:
I'm all for jokes like this, but let's make them self-deprecating rather than being jabs at dead CEOs of competitors.

Reply:
Done